enterprise/reporting/error_extractor.go

```diff
@@ -357,16 +363,28 @@ func containsDeprecationKey(errorMessage, key string) bool {
 	return strings.HasPrefix(errorMessage, key) || strings.Contains(errorMessage, " "+key)
 }
 
-func (ext *ExtractorHandle) isVersionDeprecationError(errorMessage string) bool {
-	// Normalize error message
-	cleanedError := strings.ReplaceAll(strings.ToLower(errorMessage), "-", " ")
-	return slices.ContainsFunc(deprecationKeywordSets, func(keywordSet []string) bool {
+func containsAllKeywords(errorMessage string, keywordSets [][]string) bool {
+	return slices.ContainsFunc(keywordSets, func(keywordSet []string) bool {
 		return !slices.ContainsFunc(keywordSet, func(keyword string) bool {
-			return !containsDeprecationKey(cleanedError, keyword)
+			return !containsDeprecationKey(errorMessage, keyword)
 		})
 	})
 }
 
+func (ext *ExtractorHandle) isVersionDeprecationError(errorMessage string) bool {
+	// Normalize error message
+	cleanedError := strings.ReplaceAll(strings.ToLower(errorMessage), "-", " ")
+	for key, keywordSets := range deprecationKeywordSets {
+		if !containsDeprecationKey(cleanedError, key) {
+			continue
+		}
+		if containsAllKeywords(cleanedError, keywordSets) {
+			return true
+		}
```

Comment:
[NItpick] How frequently can `containsAllKeyworkds` run ? We are searching for individual patterns in loop. This can be costly. We should have a look at pyroscope cpu profiles after release

FYI, we are considering Aho-corasick for bot pattern matching which can be used here too.

Reply:
This is good suggestion but I need to match if word is at the start of message or it have space before, that is hard to do using Aho-corasick matcher